source/extensions/filters/http/grpc_stats/grpc_stats_filter.cc

```diff
@@ -36,24 +34,16 @@ class GrpcStatsFilter : public Http::StreamFilter {
     if (grpc_request_) {
       uint64_t delta = request_counter_.decode(data);
       if (delta > 0) {
+        if (emit_filter_state_) {
```

Comment:
nit: just move this guard into the `maybeWrite` function. same below.

Reply:
Done.